drivers/net/ethernet/microchip/enc28j60.c

```diff
@@ -1080,7 +1080,7 @@ static void enc28j60_tx_clear(struct net_device *ndev, bool err)
 		if (!err)
 			ndev->stats.tx_bytes += priv->tx_skb->len;
 		dev_kfree_skb(priv->tx_skb);
-		priv->tx_skb = NULL;
+		smp_store_mb(priv->tx_skb, NULL);
```

Comment:
Here we want to make sure that we read the actual value of priv->tx_skb. We dont care about the NULL written to tx_skb, since nobody reads this value in another thread. So to ensure that we read the actual value of priv->tx_skb  we need a SMP read memory barrier _before_ we access it: the barrier ensures that all values cached at that time are discarded. Thus the processor is forced to read the value for priv->tx_skb from memory.

So 
smp_rmb();
if (priv->tx_skb) ...